tensorflow/compiler/mlir/xla/transforms/xla_legalize_to_linalg.cc

```diff
@@ -192,6 +192,107 @@ class ScalarPointwiseToStandardConverter : public OpConversionPattern<LhloOp> {
   }
 };
 
+//===----------------------------------------------------------------------===//
+// xla_lhlo.convolution conversion pattern.
+//===----------------------------------------------------------------------===//
+
+/// Converts xla_lhlo.convolution operation to a linalg.conv op.
+struct ConvToLinalgConverter : public OpConversionPattern<xla_lhlo::ConvOp> {
+ public:
+  using OpConversionPattern<xla_lhlo::ConvOp>::OpConversionPattern;
+
+  //  This code has been adapted from IREE's
+  //  (https://github.com/google/iree/) xla_hlo -> linalg conversion.
+  LogicalResult matchAndRewrite(
+      xla_lhlo::ConvOp op, ArrayRef<Value> args,
+      ConversionPatternRewriter& rewriter) const final {
+    if (const auto& dimensionNumbers = op.dimension_numbers()) {
```

Comment:
Can you spell the type instead of `auto` here? If it is an attribute you should be able to take it by value an not reference

Reply:
Done. It's actually a large struct - `xla_lhlo::ConvDimensionNumbers` - holding several other attributes; so by reference.